src/encoder.rs

```diff
@@ -350,6 +356,89 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  /// Computes the variance of 8x8 blocks factoring Contrast Sensitivity
+  /// Function in the tx_domain.
+  ///
+  /// # Panics
+  ///
+  /// Panics if the activity mask is not decimated by 8 on both axes.
+  pub fn compute_activity_mask(&mut self) {
+
+    // Contrast Sensiticity Function
+    let csf: [[f32; 8]; 8] = [
+      [1.608_443, 2.339_554, 2.573_509, 1.608_443, 1.072_295, 0.643_377, 0.504_610, 0.421_887],
+      [2.144_591, 2.144_591, 1.838_221, 1.354_478, 0.989_811, 0.443_708, 0.428_918, 0.467_911],
+      [1.838_221, 1.979_622, 1.608_443, 1.072_295, 0.643_377, 0.451_493, 0.372_972, 0.459_555],
+      [1.838_221, 1.513_829, 1.169_777, 0.887_417, 0.504_610, 0.295_806, 0.321_689, 0.415_082],
+      [1.429_727, 1.169_777, 0.695_543, 0.459_555, 0.378_457, 0.236_102, 0.249_855, 0.334_222],
+      [1.072_295, 0.735_288, 0.467_911, 0.402_111, 0.317_717, 0.247_453, 0.227_744, 0.279_729],
+      [0.525_206, 0.402_111, 0.329_937, 0.295_806, 0.249_855, 0.212_687, 0.214_459, 0.254_803],
+      [0.357_432, 0.279_729, 0.270_896, 0.262_603, 0.229_778, 0.257_351, 0.249_855, 0.259_950]];
+    
+    let luma_plane = &self.input.planes[0];
+    let PlaneConfig { width, height, .. } = luma_plane.cfg;
+    let activity_plane = &mut self.activity_mask;
+    let PlaneConfig { xdec, ydec, .. } = activity_plane.cfg;
+    let activity_mask: &mut [u16] = &mut *activity_plane.data;
+    assert!(xdec == 3 && ydec == 3);
+    let bit_depth = self.bit_depth;
+
+    let aligned_luma = Rect {
+      x: 0_isize,
+      y: 0_isize,
+      width: (width >> xdec) << xdec,
+      height: (height >> ydec) << ydec,
+    };
+    let luma = PlaneRegion::new(luma_plane, aligned_luma);
+
+    for x in 0 .. width >> xdec {
+      for y in 0 .. height >> ydec {
+
+        let block_rect = Area::Rect{
+          x: (x << xdec) as isize,
+          y: (y << ydec) as isize,
+          width: 8,
+          height: 8,
+        };
+        
+        let block = luma.subregion(block_rect);
+        
+        let mut mean: i32 = block.rows_iter().flatten().map(|&val| {let int: i32 = CastFromPrimitive::cast_from(val); int}).sum();
+        mean >>= xdec + ydec;
```

Comment:
We may skip subtracting the mean by directly zeroing the DC value after transforming to the frequency domain.

Reply:
How about I change the DC coeff multiplication factor of the CSF to 0?